tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs

```diff
@@ -1080,6 +1080,59 @@ public void EnsurePublicApiSurfaceExistsOnInterface()
             PublicClientApplicationTests.CheckBuilderCommonMethods(byRefreshTokenBuilder);
         }
 
+        [TestMethod]
+        public async Task ConfidentialClientSuggestedExpiryAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                              .WithRedirectUri(TestConstants.RedirectUri)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              .BuildConcrete();
+
+                long expectedTimeDiff = 32800;
+                long shortExpectedTimeDiff = 30800;
+
+                //Since the timeDiff is calculated using the DateTimeOffset.Now, there may be a slight variation of a few seconds depending on test environment.
+                int allowedTimeVariation = 10;
+
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage(TestConstants.DefaultAccessToken, expectedTimeDiff.ToString());
+
+                TokenCacheHelper.PopulateDefaultAppTokenCache(app);
+                app.AppTokenCache.SetAfterAccess((args) => 
+                {
+                    if (args.HasStateChanged == true)
+                    {
+                        Assert.IsNotNull(args.SuggestedCacheExpiry);
+                        var timeDiff = args.SuggestedCacheExpiry.Value.ToUnixTimeSeconds() - DateTimeOffset.Now.ToUnixTimeSeconds();
+                        Assert.IsTrue(Enumerable.Range((int)expectedTimeDiff - allowedTimeVariation, (int)expectedTimeDiff).Contains((int)expectedTimeDiff) );
```

Comment:
This looks wrong. Enumerable.Range will generate all int numbers between N and M. Whean dealing with time, it's easier to use `TimeSpan`. You already have a helper like: 

`AreEqual(DateTimeOffset expected, DateTimeOffset actual, TimeSpan delta)`

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/6a54e7df0f005bab9138f70399007e9d07d45473/tests/Microsoft.Identity.Test.Common/Core/Helpers/CoreAssert.cs#L35